frontend/src/components/ViewTrips/trips-container.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import Trip from './trip';
+
+import * as DATABASE from '../../constants/database.js';
+
+
+/**
+ * Returns a promise of a query object containg the array of Trip Documents
+ * corresponding to the trips that the current user is a collaborator on.
+ *
+ * @param {firebase.firestore.Firestore} db The Firestore database instance.
+ * @param {string} userEmail The email corresponding to the current user
+ *    logged in.
+ * @return {Promise<!firebase.firestore.QuerySnapshot>} Promise object
+ *    containing the query results with zero or more Trip  documents.
+ */
+function queryUserTrips(db, userEmail) {
+  return db.collection(DATABASE.COLLECTION_TRIPS)
+      .where(DATABASE.TRIPS_COLLABORATORS, 'array-contains', userEmail)
+      .get();
+}
+
+/**
+ * Grabs Trips query result from `queryUserTrips()` and returns an array of
+ * `<Trip>` elements as defined in `trip.js`.
+ *
+ * @param {Promise<!firebase.firestore.QuerySnapshot>} querySnapshot Promise
+ *    object containing the query results with zero or more Trip  documents.
+ * @return {Promise<!Array<Trip>>} Promise object containing an array
+ *    of Trip React/HTML elements corresponding to the Trip docsuments included
+ *    in 'querySnapshot`.
+ */
+function serveTrips(querySnapshot) {
+  return new Promise(function(resolve) {
+    const tripsContainer = querySnapshot.docs.map(doc =>
+        ( <Trip key={doc.id} tripObj={doc.data()} tripId={doc.id} /> ));
+    resolve(tripsContainer);
+  });
+}
+
+/**
+ * Returns a `<div>` element with the specified error message.
+ *
+ * @param {string} error Error message in `componentDidMount()` catch statement.
+ * @return {Promise<HTMLDivElement>} Promise object containing a `<div>` element
+ *    with the error message `error` inside.
+ */
+function getErrorElement(error) {
+  return new Promise(function(resolve) {
+    console.log(`Error in Trips Container: ${error}`);
+    resolve(( <div><p>Error: Unable to load your trips.</p></div> ));
+  });
+}
+
+/**
+ * Component corresponding to the container containing a users trips.
+ *
+ * @extends React.Component
+ */
+class TripsContainer extends React.Component {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+    this.state = {trips: []};
+  }
+
+  /** @inheritdoc */
+  async componentDidMount() {
+    try {
+      const querySnapshot = await queryUserTrips(
+          this.props.db, this.props.userEmail);
```

Comment:
Are these fields always guaranteed to be set?

Reply:
Since this component is called in the index file with these arguments, they should be set.

That being said, it might be a good idea to document this component with the props you expect to receive, namely the db and userEmail.

Reply:
Theoretically it is possible that the firestore instance variable is null (maybe). My error handling in componentDidMount() was able to handle this case and returned the expected error message. Is this something I should have in a test case if possible?

The user email will be retrieved from the authentication utils. So if there are any hiccups there, it should probably be handled within the utils and kick the user out of the app (or something similar). But I can ask @keiffer01 to confirm.

Reply:
On the topic of the user email, yes any error that happens with that function is handled on the authentication side. If the user's email returns null, it means somehow they managed to get onto the view-trips page without being authenticated first. Since that is very bad, they will immediately be kicked back to the sign-in screen. So you shouldn't have to worry about error handling this function on your side.

Reply:
If you handle errors in your function then you should probably add a test for it, yes.

Reply:
That is what I thought. However I believe this test would fall under integration tests since it would require having access to the Firestore instance and this function is only run when the component is called (componentDidMount).